osquery/core/windows/process.cpp

```diff
@@ -0,0 +1,221 @@
+  id_ = duplicateHandle(process.nativeHandle());
+  return *this;
+}
+
+bool PlatformProcess::operator==(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) == ::GetProcessId(process.nativeHandle()));
+}
+
+bool PlatformProcess::operator!=(const PlatformProcess& process) const {
+  return (::GetProcessId(nativeHandle()) != ::GetProcessId(process.nativeHandle()));
+}
+
+int PlatformProcess::pid() const {
+  return ::GetProcessId(id_);
+}
+
+bool PlatformProcess::kill() const {
+  if (id_ == kInvalidPid) {
+    return false;
+  }
+  
+  return ::TerminateProcess(id_, 0);
+}
+
+PlatformProcess PlatformProcess::launchWorker(const std::string& exec_path, const std::string& name) {
+  ::STARTUPINFOA si = { 0 };
+  ::PROCESS_INFORMATION pi = { 0 };
+  
+  si.cb = sizeof(si);
+  
+  // TODO(#1991): We currently do not sanitize or check for bad characters in for the worker name. Names
+  //              with double quotes have the potential of causing argument parsing issues. However, it
+  //              is not a huge concern for the worker process, for it does not pass any command line 
+  //              arguments.
+  std::stringstream argv_stream;
+  argv_stream << "\"" << name << "\"";
+  
+  std::stringstream handle_stream;
+  
+  // TODO(#1991): The HANDLE exposed to the child process only has SYNCHRONIZE and 
+  //              PROCESS_QUERY_LIMITED_INFORMATION privileges which is enough to cover the current use 
+  //              cases. This may not be the case in the future...
+  HANDLE hLauncherProcess = ::OpenProcess(SYNCHRONIZE | PROCESS_QUERY_LIMITED_INFORMATION, 
+                                          TRUE, 
+                                          GetCurrentProcessId());
+  if (hLauncherProcess == NULL) {
+    return PlatformProcess(kInvalidPid);
+  }
+  
+  handle_stream << hLauncherProcess;
+  std::string handle = handle_stream.str();
+  
+  // In the POSIX version, the environment variable OSQUERY_WORKER is set to the string form of the
+  // child process' process ID. However, this is not easily doable on Windows. Since the value does
+  // not appear to be used by the rest of osquery, we currently just set it to '1'.
+  //
+  // For the worker case, we also set another environment variable, OSQUERY_LAUNCHER. OSQUERY_LAUNCHER
+  // stores the string form of a HANDLE to the current process. This is mostly used for detecting the 
+  // death of the launcher process in WatcherWatcherRunner::start
+  if (!setEnvVar("OSQUERY_WORKER", "1") ||
+      !setEnvVar("OSQUERY_LAUNCHER", handle.c_str())) {
+    ::CloseHandle(hLauncherProcess);
+
+    return PlatformProcess(kInvalidPid);
+  }
+  
+  // We don't directly use argv.c_str() as the value for lpCommandLine in CreateProcess since
+  // that argument requires a modifiable buffer. So, instead, we off-load the contents of argv
+  // into a vector which will have its backing memory as modifiable.
+  std::string argv = argv_stream.str();
```

Comment:
Why have a `std::streamstream`? Can you not just initialize a `auto argv = std::string("\"") + name + "\"";`?